connect-inject/container_init.go

```diff
@@ -81,6 +82,13 @@ func (h *Handler) containerInit(pod *corev1.Pod, k8sNamespace string) (corev1.Co
 		panic("No service found. This should be impossible since we default it.")
 	}
 
+	// If ACLs are enabled, the ACL token is assumed to be in the ServiceAccount that
+	// shares the name with the Service. If the name of the service does not match that
+	// of the ServiceAccount, the token will not be found leading to an opaque 403 error.
```

Comment:
Can you reword this to say that the ACL token returned from `consul login` is only valid for a service with the same name as the serviceAccountName. I think the "assumed to be in the ServiceAccount" doesn't fully capture what's going on because the ACL token isn't in the service account. We exchange the service account token for a consul ACL token via `consul login`.